app/scripts/modules/netflix/canary/canaryConfigSelector.directive.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+
+let angular = require('angular');
+
+module.exports = angular
+  .module('spinnaker.netfilx.widgits.canaryConfigSelector.directive', [
+    require('./canary.read.service'),
+    require('../../core/utils/lodash')
+  ])
+  .directive('canaryConfigSelector', () => {
+    return {
+      scope: {},
+      bindToController: {
+        model: '=' ,
+        applicationName: '='
+      },
+      templateUrl: require('./canaryConfigSelector.directive.html'),
+      controllerAs: 'vm',
+      controller: function($scope, canaryReadService, _) {
+        let vm = this;
+
+        vm.selectorChanged = (model) => {
+          if (model) {
+            vm.model = model;
+            vm.model.successfulScore = model.canarySuccessCriteria.canaryResultScore;
+            vm.model.unhealthyScore = model.canaryHealthCheckHandler.minimumCanaryResultScore;
+            vm.model.resultStrategy = model.combinedCanaryResultStrategy;
+          }
+        };
+
+        vm.isNameInConfigList = (selectedName) => {
+          return getSelected(vm.canaryConfigs, selectedName).length ? true : false;
+        };
+
+        let setOptions = (results) => {
+          vm.canaryConfigs = results;
+          return vm.canaryConfigs;
+        };
+
+        let getSelected = _.curry((canaryConfigs = [], canaryName) => {
+          return canaryConfigs.filter((config) => config.name === canaryName);
+        });
+
+        let findConfigAndSet = _.flowRight(vm.selectorChanged, _.first, getSelected);
```

Comment:
It is more inline with Functional Programming idiom of a 'compose' function.  Lodash is not super FP friendly so I'm using what I've got.  

Sent from Outlook Mobile




On Wed, Feb 3, 2016 at 12:19 PM -0800, "Chris Berry" <notifications@github.com> wrote:












In app/scripts/modules/netflix/canary/canaryConfigSelector.directive.js:
> +        };
> +
> +        vm.isNameInConfigList = (selectedName) => {
> +          return getSelected(vm.canaryConfigs, selectedName).length ? true : false;
> +        };
> +
> +        let setOptions = (results) => {
> +          vm.canaryConfigs = results;
> +          return vm.canaryConfigs;
> +        };
> +
> +        let getSelected = _.curry((canaryConfigs = [], canaryName) => {
> +          return canaryConfigs.filter((config) => config.name === canaryName);
> +        });
> +
> +        let findConfigAndSet = _.flowRight(vm.selectorChanged, _.first, getSelected);



why flowRight instead of flow?



—
Reply to this email directly or view it on GitHub.